doozerlib/assembly_inspector.py

```diff
@@ -63,12 +76,52 @@ def does_permit(self, issue: AssemblyIssue) -> bool:
                     return True
         return False
 
+    def _check_installed_packages_for_rpm_delivery(self, component: str, component_description: str, rpm_packages: Dict[str, Dict[str, Any]]):
+        """
+        If rpm_deliveries is configured, this checks installed packages in a component
+        follow the stop-ship/ship-ok tagging requirement.
+        See https://issues.redhat.com/browse/ART-6100 for more information.
+
+        :param component: Component name, like `rhcos` or image distgit key
+        :param component_description: Component description, used to include more details for the component in the AssemblyIssue message
+        :param rpm_packages: A dict; key is rpm package name, value is brew build dict
+        """
+        self.runtime.logger.info("Checking installed rpms in %s", component_description)
+        issues: List[AssemblyIssue] = []
+        for package_name, rpm_build in rpm_packages.items():
+            if package_name in self._rpm_deliveries:
+                rpm_delivery_config = self._rpm_deliveries[package_name]
+                self.runtime.logger.info("Getting tags for rpm build %s...", rpm_build['nvr'])
+                tag_names = {tag["name"] for tag in self.brew_session.listTags(build=rpm_build["id"])}
```

Comment:
feels like it might be a bit slow to look these tags up for every entry that has a listed RPM. could be a cached function?

Reply:
Yes, I will cache this in the next push.